### PR TITLE
Implements writeFileSync

### LINKED ISSUE #415: Implement deno.readFileSync and deno.writeFileSync
Here's a pointer to the prototype implementation: 
https://github.com/ry/deno/blob/golang/os.go#L171-L195

### PR DESCRIPTION
**Pull request by ry**
Fixes #415

### CODE REVIEW

File: js/unit_tests.ts [hunk 1]
```diff
@@ -107,6 +108,35 @@ test(function tests_readFileSync_NotFound() {
...
+  const enc = new TextEncoder();
+  const data = enc.encode("Hello");
+  const filename = "/baddir/test.txt";
+  // The following should fail because /baddir doesn't exist (hopefully).
+  let caughtError = false;
+  try {
+    deno.writeFileSync(filename, data);
+  } catch (e) {
+    caughtError = true;
+    // TODO assertEqual(e, deno.NotFound);
```
Comment (piscisaureus) [inline js/unit_tests.ts]: I don't understand. You want to have a "const" error object for all NotFound error objects?
I'd rather have see relevant information (such as the failing path) as properties on the object.

File: src/handlers.rs [hunk 2]
```diff
@@ -413,6 +421,28 @@ fn handle_read_file_sync(
...
+  d: *const DenoC,
+  builder: &mut FlatBufferBuilder,
+  filename: &str,
+  data: &[u8],
+  perm: u32,
+) -> HandlerResult {
+  debug!("handle_write_file_sync {}", filename);
+  let deno = from_c(d);
+  if deno.flags.allow_write {
+    // TODO(ry) Use perm.
```
Comment (piscisaureus) [inline src/handlers.rs]: Yeah that's kinda bad. Maybe `assert(perm & 0o666 === 0o666)` then.

Comment (NewLunarFire) [reply src/handlers.rs]: https://doc.rust-lang.org/nightly/std/fs/struct.Permissions.html#method.readonly

Comment (ry) [reply src/handlers.rs]: Thanks - could you submit a PR?

Comment (NewLunarFire) [reply src/handlers.rs]: #567

**Pull request merged**